### PR TITLE
Feature/73 set current location as start location

### PR DESCRIPTION
**Pull request by dkrakau**
### Description

Issue: #73 feature/73-set-current-location-as-start-location

## Checklist

- [x] My code follows the style guidelines
- [ ] I have documented my code
- [x] My changes generate no new warnings
- [x] I have linked the Issue
- [x] The Pull Request covers all the Checkpoints from the tagged Issue
- [x] I have noted the hours of work i have put in the issue

## I have edited following Parts of the project

- [x] Frontend
- [ ] Backend
- [ ] Github
- [ ] Root

## If tests are required did you make them?

- [ ] No tests required
- [x] Yes
- [ ] No

### CODE REVIEW

File: frontend/src/components/LocationSearch/CurrentLocationButton.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import "leaflet/dist/leaflet.css";
+import { Location } from "../../api";
+import { useCurrentLocation } from "../../hooks/useCurrentLocation";
+import { useLocationFinderApi } from "../../services/apiClients/ApiClientsContext";
+
+export interface CurrentLocationButtonProps {
+  onButtonClicked: (location: Location) => void;
+}
+
+export function CurrnetLocationButton(props: CurrentLocationButtonProps): JSX.Element {
```
Comment (Paulomart) [inline frontend/src/components/LocationSearch/CurrentLocationButton.tsx:11]: typo: 
```suggestion
export function CurrentLocationButton(props: CurrentLocationButtonProps): JSX.Element {
```

File: frontend/src/components/LocationSearch/CurrentLocationButton.tsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+      latitude: currentLocation.location.coords.latitude,
+      longitude: currentLocation.location.coords.longitude
+    });
+
+    props.onButtonClicked({
+      id: locationWithAssinedStops.id,
+      name: locationWithAssinedStops.name,
+      details: locationWithAssinedStops.details,
+      type: locationWithAssinedStops.type
+    });
```
Comment (Paulomart) [inline frontend/src/components/LocationSearch/CurrentLocationButton.tsx:33]: Hier fehlta auch noch komplett die Fehlerbehandlung wenn 

```
const locationWithAssinedStops = await api.locationFinderControllerFindLocationsAtCoordinates({
```

fehlschlägt.

**Review by Paulomart — APPROVED** [review]

File: frontend/src/components/LocationSearch/CurrentLocationButton.tsx (hunk 1 above)
Comment (dkrakau) [reply frontend/src/components/LocationSearch/CurrentLocationButton.tsx]: Oh ok, i have corrected it.

File: frontend/src/components/LocationSearch/CurrentLocationButton.tsx (hunk 2 above)
Comment (dkrakau) [reply frontend/src/components/LocationSearch/CurrentLocationButton.tsx]: Exception handling is implemented.

**Pull request merged**